### DISCUSSION TITLE
新建存储时报错Failed create storage in database: database disk image is malformed

### DISCUSSION THREAD

**Discussion by Sunset-go** [AlistGo/alist#7320] [Q&A]
在新建123云盘、联通云盘、阿里云盘的存储时出现"Failed create storage in database: database disk image is malformed"的错误，不知道哪里出问题，求解大佬！！！

**Comment by ZZ0YY** [comment]
> 在新建123云盘、联通云盘、阿里云盘的存储时出现"Failed create storage in database: database disk image is malformed"的错误，不知道哪里出问题，求解大佬！！！

你这啥都不提供 谁知道怎么解决 就贴个AI给你看 你这问法也就配 ai 回答了
这个错误通常是由于SQLite数据库文件损坏引起的。以下是一些可能的解决方法：

1. **重启服务**：有时候简单的重启Alist服务可以解决问题。你可以尝试以下命令：
    ```bash
    systemctl stop alist
    systemctl start alist
    ```

2. **修复数据库**：如果重启服务无效，可以尝试修复SQLite数据库。首先备份你的数据库文件，然后使用以下命令修复：
    ```bash
    sqlite3 /path/to/your/database.db "PRAGMA integrity_check;"
    ```
    如果检查结果显示数据库损坏，可以尝试导出和重新导入数据库：
    ```bash
    sqlite3 /path/to/your/database.db .dump > backup.sql
    sqlite3 /path/to/your/new_database.db < backup.sql
    ```

3. **替换损坏的数据库文件**：如果你有之前的数据库备份，可以直接替换损坏的数据库文件。

4. **检查文件系统**：有时候文件系统的问题也会导致数据库损坏。可以检查磁盘是否有错误并修复：
    ```bash
    fsck /dev/sdX
    ```

如果以上方法都无法解决问题，建议查看Alist的官方文档或社区讨论，可能会有更多针对性的解决方案¹²³。

希望这些方法能帮到你！如果还有其他问题，随时告诉我。😊

¹: [GitHub讨论](https://github.com/alist-org/alist/discussions/3109)
²: [GitHub问题](https://github.com/alist-org/alist/issues/2821)
³: [CSDN博客](https://blog.csdn.net/weixin_42585386/article/details/108822224)

源: 与 Copilot 的对话， 2024/10/12
(1) 新添加存储的时候 提示：Failed create storage in database: database is locked .... https://github.com/alist-org/alist/discussions/3109.
(2) Failed create storage in database: database is locked - GitHub. https://github.com/alist-org/alist/issues/2821.
(3) SQLite数据库学习笔记1:报错:The database disk image is .... https://blog.csdn.net/weixin_42585386/article/details/108822224.
(4) SVN 报错 sqlite[S11]: database disk image is malformed的 .... https://blog.csdn.net/wqh0830/article/details/104070054.
(5) 解决SQLite3数据库Error: database disk image is malformed .... https://blog.csdn.net/BankxiY/article/details/113696087.
(6) undefined. https://alist.nn.ci/zh/guide/advanced/search.html#搜索提示.
(7) undefined. https://api.nn.ci/alist/ali_open/token.

**Reply by Sunset-go** [reply]
解决了，数据库损坏，备份数据重新生成data文件夹就行
